### PR TITLE
Move BlackBox to app/lib

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Refactor?
- [X] DX

## Description

Ages ago we had a discussion about cleaning up our `app/` structure a little bit. While I don't think we can get rid of `BlackBox` just yet I don't think it needs to have its own directory.

## Added/updated tests?

- [X] Yes

## [Forem core team only] How will this change be communicated?

- [X] This change does not need to be communicated, and this is why not: internal code organization only

### CODE REVIEW

**Review by Ridhwana — APPROVED** [review]

**Review by jeremyf — APPROVED** [review]

**Review by djuber — APPROVED** [review]

**Comment by djuber** [comment]
`.simplecov` has a filter added for this removed directory that we may need to update or remove?

**Comment by citizen428** [comment]
Good catch @djuber, I updated the file.

File: .simplecov [hunk 1]
```diff
@@ -1,18 +1,14 @@
 unless ENV["COVERAGE"] == "false"
   SimpleCov.start "rails" do
     add_filter "/spec/"
-    add_filter "/dashboards/"
```
Comment (citizen428) [inline .simplecov:4]: This doesn't seem to exist, so I removed it.

File: .simplecov [hunk 2]
```diff
@@ -1,18 +1,14 @@
...
     add_filter "/app/controllers/admin/"
-    add_filter "/app/black_box/"
-    add_filter "/app/fields/"
```
Comment (citizen428) [inline .simplecov:7]: We don't have `app/fields`.

File: .simplecov [hunk 3]
```diff
@@ -1,18 +1,14 @@
...
+    add_filter "/app/lib/black_box/"
```
Comment (citizen428) [inline .simplecov:5]: Update the path.

File: .simplecov [hunk 4]
```diff
@@ -1,18 +1,14 @@
...
     add_filter "/app/views/admin/"
 
-    # TODO: [thepracticaldev/oss] enable this when we upgrade to SimpleCov 0.18
```
Comment (citizen428) [inline .simplecov:10]: We're on 0.21.2 now so I enabled this.

File: .simplecov [hunk 5]
```diff
@@ -1,18 +1,14 @@
...
-    # enable_coverage :branch
+    enable_coverage :branch
 
     add_group "Decorators", "app/decorators"
     add_group "Errors", "app/errors"
-    add_group "Labor", "app/labor"
```
Comment (citizen428) [inline .simplecov:15]: `app/labor` no longer exists.

**Pull request merged**